TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -10,38 +10,40 @@ import Foundation
 class NetworkManager {
     
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
-        guard var components = URLComponents(string: urlString) else {
-            completion(nil)
-            return
-        }
-        components.queryItems = parameters.map { (key, value) in
-            URLQueryItem(name: key, value: value)
-        }
-        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
-        guard let url = components.url else {
-            completion(nil)
-            return
-        }
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "GET"
-        let urlSession = URLSession.shared
-        let fileDownloadTask = urlSession.dataTask(
-            with: urlRequest
-        ) { data, _, error in
-            let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
-                DispatchQueue.main.async {
-                    completion(loadedData)
-                }
+        DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
+            guard var components = URLComponents(string: urlString) else {
+                completion(nil)
```

Comment:
Alternative solution:
- You can record the last time when an actual request happened.
- Whenever a new request is about to happen you can check if 12.5ms has passed since that moment.
-- If the time has passed, update the time recording and make an actual request.
-- If the time has not passed, repeat the whole checking procedure later (The delay can be calculated from the last timestamp or simply use 12.5ms)